### ISSUE TITLE
Hangs while decrypting a db.

### ISSUE THREAD

**Issue by liesnikov** [Kunzisoft/KeePassDX#187]
* App version: 2.5.0.0beta17-libre
* Android version: 8.1.0
* Database settings:
  * Encryption: AES-256 (rijndael algo, according to the app)
  * Key derivation: Argonn2 (KDBX 4)
  * Transform rounds: 13
  * Memory usage: 64 mb
  * Parallelism: 4 threads

Database size: around 3 mb, no attachments as far as i remember.

As in the title: it passes "creating database key" stage and then just hangs indefinitely (>10 minutes). 
Nothing similar in KeePassXC. Resolved it by deleting permanently (from recycle bin) a couple of small entries (no notes and attachments) and folders.

Can't provide files for testing, but have both versions and may find some time to test fixes or provide logs.

**Comment by J-Jamet** [comment]
I don't really see what it could be. It's a weird bug. If you can look at the logs and see if there are any exception it would be nice.

**Comment by liesnikov** [comment]
nothing in logcat, unfortunately:
```
09-21 10:57:19.638   788  3059 I ActivityManager: START u0 {act=android.intent.action.VIEW dat=content://com.dropbox.android.FileCache/filecache/<some-file-cache> typ=application/octet-stream flg=0x10000003 cmp=com.kunzisoft.keepass.libre/com.kunzisoft.keepass.password.PasswordActivity (has extras)} from uid 10129
09-21 10:57:19.708   788  1793 W WindowManager: TaskSnapshotSurface.create: Failed to find main window for token=AppWindowToken{d73acd2 token=Token{a819b5d ActivityRecord{ab76c07 u0 com.kunzisoft.keepass.libre/com.kunzisoft.keepass.password.PasswordActivity t6089}}}
...
09-21 10:59:16.760   788   818 I ActivityManager: Force stopping com.kunzisoft.keepass.libre appid=10125 user=0: from pid 788
09-21 10:59:16.760   788   818 I ActivityManager: Killing 25295:com.kunzisoft.keepass.libre/u0a125 (adj 100): stop com.kunzisoft.keepass.libre
09-21 10:59:16.762   788   818 W ActivityManager: Force removing ActivityRecord{ab76c07 u0 com.kunzisoft.keepass.libre/com.kunzisoft.keepass.password.PasswordActivity t6089}: app died, no saved state
09-21 10:59:16.781   788   818 I ActivityManager:   Force finishing activity ActivityRecord{f4e6f0f u0 com.kunzisoft.keepass.libre/com.kunzisoft.keepass.password.PasswordActivity t6089}
```

**Comment by NTICompass** [comment]
I'm having this issue.  When I try to open my `.kdbx` file, it gets stuck at "Decrypting database content...".

My KeePass file uses ChaCha20 and Argon2 (25 rounds, 64 MiB memory and 4 threads according to KeepassXC).  It was able to be opened up until a few days ago, and all I'd changed was adding in a new website/password.

**Comment by Horlacher** [comment]
Same strange bug, maybe this could help with debugging:
After changing the key derivation from AES-KDF to Argon2 in KeePassXC 2.4.1 I've got the freeze in KeePassDX 2.5.0.0beta18. It only happened in a database with many entries, not in a big DB with file attachments.

Workarounds:
1. Changed Memory usage from 128mb to 64mb and now it works, even after setting it back to 128mb - I wonder for how long
2. Removing many entries - though after re-importing them, KeePassDX fails again
3. Using another app

Creating a new DB and importing the entries in KeePassXC from the old DB did not solve the problem.

**Comment by wassname** [comment]
The workaround above didn't work (changing rounds, memory usage). Smaller test databases also work for me. What did work on my large database as changing to AES-KDF (KDBX 4). 

What a weird bug.

**Comment by J-Jamet** [comment]
The problem comes from the fragment that displays the dialog.
In older versions, the database was loaded from an activity. Now (in version 2.5RC2), the application uses a service created for loading data. There is still a bug with the dialog fragment which displays the opening state, but it is greatly minimized.
And it is possible to remove this loading dialog by putting the application in the background; if the loading is finished, the dialog will be auto deleted.
I will try to further improve the loading by using the coroutines which support lifecycles.

**Comment by J-Jamet** [comment]
Users still have problems in version 2.5 stable, this does not affect the data but it is visually annoying. I will add a longer minimum opening time to find a lasting solution.

**Comment by J-Jamet** [comment]
The problem is solved in version 2.6
